### PR TITLE
Refactor discovery

### PR DESCRIPTION
**Pull request by Darsstar**
First commit:
- test with 3.11 and 3.12
- require combined coverage from all versions to be 100% instead of individual Python versions, this allows `sys.version_info < (3, 10)` checks to be used in the code base.

Second commit:
- The main purpose is to speed up discovery.
- But is also provides similair functionality to #144.
  - The difference is this PR does not use the class name, but creates names entry points instead.
  - This PR allows for a list, instead of a single class/model. The Home Assistant integration could add a config entry to manage which inverter classes/models are enabled. During discovery it would then only pass those enabled inverter classes/models to `solax.discover()`
    - `solax.discover()` can also return a list of all inverter models/classes instead of returning the fist that succeeded.
    Which can be used by Home Assistant to have the user resolve such situations.
 
Third commit:
- Provides functionality to fix https://github.com/home-assistant/core/pull/107615 properly
  - x3_hybrid_g4 has a monlty reseting energy, which isn't supported by `Measurement`, but that can easily be added if desired.

Fourth commit:
- Adds inverter SN to the named tuple. Home Assistant could then use that instead of the dongle SN. Which would have the benefit of having only one entity if a broken dongle needs to be replaced.

@squishykid It took me longer to finish than expected. Hope you enjoyed your holiday!
PS. I assume you are the one who started following me on Trakt :p

### CODE REVIEW

**Comment by brew-your-own** [comment]
Nice!
Do you also have patches for HA to apply your approach in the component?

**Comment by squishykid** [comment]
looks good to me! nice work mate.

File: solax/inverters/x1_hybrid_gen4.py [hunk 1]
```diff
@@ -15,28 +16,20 @@ class X1HybridGen4(Inverter):
                 "sn",
             ): str,
             vol.Required("ver"): str,
-            vol.Required("Data"): vol.Schema(
+            vol.Required("data"): vol.Schema(
```
Comment (nazar-pc) [inline solax/inverters/x1_hybrid_gen4.py:19]: It is `Data`, not `data` in JSON. Just like `Information` below. Why did you change it?

Comment (Darsstar) [reply solax/inverters/x1_hybrid_gen4.py:19]: Because I lowercased all keys in a previous PR.

Comment (Darsstar) [reply solax/inverters/x1_hybrid_gen4.py:19]: I accedentilly hit "Comment" in the mobile app...

I meant to write `commit` not `PR`, but both are wrong, [it was in the same commit](https://github.com/squishykid/solax/pull/145/commits/4c440632e5ba9cf9ebf6403708453c4cbca1214c#diff-0b94a8b014d1902625d8bd9f0ef7e94be86fd081927dd26e0c74e777572bfd59R109).

Anyway, I assume https://github.com/squishykid/solax/issues/161 is why you decided to comment.

Comment (nazar-pc) [reply solax/inverters/x1_hybrid_gen4.py:19]: Yes, I was going through recent changes to see if something broke, but then remembered about firmware upgrade and found the root cause
